Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -1763,6 +1764,18 @@ void qMRMLSegmentEditorWidget::onMRMLSceneEndCloseEvent()
   this->updateWidgetFromMRML();
 }
 
+//-----------------------------------------------------------------------------
+void qMRMLSegmentEditorWidget::onMRMLSceneEndBatchProcessEvent()
+{
+  Q_D(qMRMLSegmentEditorWidget);
+  if (!this->mrmlScene())
+    {
+    return;
+    }
+  // force update (clear GUI if no node is selected anymore)
```

Comment:
End result of deleting the current segmentation node must be the same regardless of batch processing was activated or not.

Since the current behavior is to choose some other segmentation node when the current segmentation node gets deleted, this same behavior should be preserved here. You may need to set the current segmentation node to some other existing node (the same way how it is set when batch processing is not activated, just skipping all the intermediate node selection changes).

Reply:
@lassoan I'm not sure I understand. As this PR currently does, it maintains the behavior of when the current segmentation node is removed, it selects the last available index of the combobox which is some other other segmentation node if there was more than 1 originally in the scene.

Reply:
> it selects the last available index of the combobox which is some other other segmentation node if there was more than 1 originally in the scene.

Perfect, I just wanted to confirm that this is the behavior.

Reply:
I've checked the behavior on my computer and I've found that it was inconsistent in batch and non-batch modes, similarly to the volume selection.

How to reproduce:

- Load [this scene](https://1drv.ms/u/s!Arm_AFxB9yqHyJd5y3S3XT5BShwTvw?e=lZRaZk) (contains 3 segmentation nodes, each with a different master volume)
- Go to Segment editor module
- Run this script:

```
scene = slicer.mrmlScene
scene.StartState(scene.BatchProcessState)
slicer.mrmlScene.RemoveNode(getNode('Segmentation'))
scene.EndState(scene.BatchProcessState)
```

Result: segmentation node is now `Select a segmentation`

- Restart Slicer, load the same scene
- Go to Segment editor module
- Run this script:

```
scene = slicer.mrmlScene
slicer.mrmlScene.RemoveNode(getNode('Segmentation'))
```

Result: selected segmentation node is now `Segmentation_1`

The problem may be similar here as above, that `setSegmentationNode` has been completely missed. I don't think we can skip it, we can just delay it until the batch processing is completed.

There is an early return from `onSegmentationNodeChanged`, but there are still updates due to `updateWidgetFromMRML` called due to node reference change in segment editor parameter node.

We could change the current behavior to always unselect the segmentation (in both batch and non-batch mode), but the fix should not introduce any difference between batch and non-batch mode.

Reply:
ah thanks for the check. I will make sure the segmentation node and master volume node are appropriately set at the end of the process. I had been meaning to confirm that after my assumption that `updateWidgetFromMRML` would've completed those actions.